server/activity_manager.go

```diff
@@ -71,13 +72,17 @@ func (m *ActivityManager) CreateActivity(ctx context.Context, create *api.Activi
 		projectFind := &api.ProjectFind{
 			ID: &meta.issue.ProjectID,
 		}
-		meta.issue.Project, err = m.s.ProjectService.FindProject(ctx, projectFind)
+		projectPlain, err := m.s.ProjectService.FindProject(ctx, projectFind)
 		if err != nil {
 			return nil, fmt.Errorf("failed to find project for posting webhook event after changing the issue status: %v, error: %w", meta.issue.Name, err)
 		}
-		if meta.issue.Project == nil {
+		if projectPlain == nil {
 			return nil, fmt.Errorf("failed to find project ID %v for posting webhook event after changing the issue status %q", meta.issue.ProjectID, meta.issue.Name)
 		}
+		// TODO(dragonly): revisit the necessity of this function to depend on ActivityMeta.
+		if err := copier.Copy(meta.issue.Project, projectPlain); err != nil {
+			panic(fmt.Sprintf("failed to copy from *api.ProjectPlain to *api.Project, %v", err))
```

Comment:
The existing panic cases are all in the init related places.

For normal code workflow, we normally don't use panic even in cases like here, where we are in a pretty bad shape.

The reason is for normal code workflow, it requires judgement to determine whether it's serious enough to be considered a panic. And that judgement is subjective. So for the sake of consistency, we opt to return error.

Reply:
Fixed.